src/components/Comments/CommentAdder.jsx

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect, useContext } from "react";
+import Button from "@mui/material/Button";
+import { useParams } from "react-router-dom";
+import { postOnArticle } from "../../apis/apis";
+import { UserContext } from "../Context/usersContext";
+
+function CommentAdder({ setComments }) {
+  const { id } = useParams();
+  const { user } = useContext(UserContext);
+  const [makeAComment, setMakeAComment] = useState("");
+  const [addNewComment, setAddNewComment] = useState({
+    username: user,
+    body: makeAComment,
+  });
+
```

Comment:
I can currently mash the post button and post loads of empty comments
I was just trolling but it would be a quite likely thing to happen if user had a slow connection.
You should add some validation to the comment and make it so the user can't post a new comment while one is in the process of being posted (you could do this by disabling the post button)
You should also reset the content of the comment box to empty after the comment has been posted

Reply:
Add an isProcessing state to conditionally deactivate button whilst posting.

Reply:
handled too short issue

Reply:
handled the setting back to empty problem too :)